pytorch_lightning/plugins/precision/deepspeed_precision.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, Union
+
+import torch
+from torch.optim import Optimizer
+
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins.precision.precision_plugin import PrecisionPlugin
+
+
+class DeepSpeedPrecisionPlugin(PrecisionPlugin):
```

Comment:
This might be a dumb question but

Why is this a PrecisionPlugin? Other than the constructor, no code in this class has anything to do directly related to precision logic

Reply:
Not a dumb question, a forced design choice for the current state of accelerators as discussed with the team.

This has nothing to do with precision, but the hooks are only available in the Precision Plugin. To get this clean, we'll need to embed the precision plugin into the training type plugin, and have general hooks be exposed to the training type plugin. This will be iterated on